magento/magento2
# MC-38931: Product URL Rewrites are not removed when product removed from website

### Description (*)

Fixes URL rewrite delete/generation on website remove/assign
### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#24184

### Manual testing scenarios (*)

1. test URL rewrite generation/delete from admin
2. test URL rewrite generation/delete by REST API requests (productRepository POST/PUT/DELETE)
3. test URL rewrite generation/delete by mass update (corn should be enabled)
4. test URL rewrite generation/delete through import 


### Questions or comments


### Contribution checklist (*)
